variables.tf

```diff
@@ -0,0 +1,206 @@
+variable "public_ip_config" {
+  type = object({
+      enabled           = bool
+      allocation_method = optional(string, "Static")
+  })
+  default = {
+    enabled = false
+  }
+  validation {
+    condition     =  contains(["Static","Dynamic"], var.public_ip_config.allocation_method)
+    error_message = "Allocation method must be Static or Dynamic"
+  }
+  description = <<-DOC
+  ```
+    enabled: Optionally select true if a public ip should be created. Defaults to false.
+    allocation_method: The allocation method of the public ip that will be created. Defaults to static.      
+  ```
+  DOC
+} 
+
+# nsg needs to be an object to use the count object in main.tf. 
+variable "nic_config" {
+  type = object({
+    private_ip  = optional(string)
+    dns_servers = optional(list(string))
+    nsg = optional(object({
+      id = string
+    }))
+  })
+  default     = {}
+  description = <<-DOC
+  ```
+    private_ip: Optioanlly specify a private ip to use. Otherwise it will  be allocated dynamically.
+    dns_servers: Optionally specify a list of dns servers for the nic.
+    nsg_id: Optinally specify the id of a network security group that will be assigned to the nic.    
+  ```
+  DOC
+}
+
+variable "enable_accelerated_networking" {
```

Comment:
if this is part of the nic, this should be part of the nic config

Reply:
Changed to inside nic config.